app/src/main/java/com/lightricks/feedexercise/data/FeedRepository.kt

```diff
@@ -1,9 +1,65 @@
 package com.lightricks.feedexercise.data
 
+import com.lightricks.feedexercise.database.FeedDatabase
+import com.lightricks.feedexercise.database.FeedItemDBEntity
+import com.lightricks.feedexercise.network.FeedApiService
+import com.lightricks.feedexercise.network.TemplatesMetadata
+import io.reactivex.Completable
+import io.reactivex.Observable
+import io.reactivex.schedulers.Schedulers
+
 /**
  * This is our data layer abstraction. Users of this class don't need to know
  * where the data actually comes from (network, database or somewhere else).
  */
 class FeedRepository {
-    //todo: implement
+    private lateinit var db: FeedDatabase
+    private lateinit var feedApiService: FeedApiService
+    private val urlPrefix = "https://assets.swishvideoapp.com/Android/demo/catalog/thumbnails/"
```

Comment:
Can you spot some duplication with the base URL?
This is something that is likely to change in the future.
It's usually better to change in one place than in many.